cns/restserver/nodesubnet_test.go

```diff
@@ -0,0 +1,78 @@
+package restserver_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/Azure/azure-container-networking/cns"
+	"github.com/Azure/azure-container-networking/cns/logger"
+	"github.com/Azure/azure-container-networking/cns/restserver"
+)
+
+// Mock implementation of PodInfoByIPProvider
+type MockPodInfoByIPProvider struct{}
+
+func (m *MockPodInfoByIPProvider) PodInfoByIP() (res map[string]cns.PodInfo, err error) {
+	return res, nil
+}
+
+// Mock implementation of CNIConflistGenerator
+type MockCNIConflistGenerator struct {
+	GenerateCalled chan bool
+}
+
+func (m *MockCNIConflistGenerator) Generate() error {
+	m.GenerateCalled <- true
+	return nil
+}
+
+func (m *MockCNIConflistGenerator) Close() error {
+	// Implement the Close method logic here if needed
+	return nil
+}
+
+func TestNodeSubnet(t *testing.T) {
+	mockPodInfoProvider := new(MockPodInfoByIPProvider)
+
+	// Create a real HTTPRestService object
+	mockCNIConflistGenerator := &MockCNIConflistGenerator{
+		GenerateCalled: make(chan bool),
+	}
+	service := restserver.GetRestServiceObjectForNodeSubnetTest(mockCNIConflistGenerator)
+	defer service.Service.Uninitialize()
```

Comment:
You might consider using `testing.(*T).Cleanup` for these

Reply:
Done.